File: src/main/java/cs/android/json/CSJSONString.java
package cs.android.json;

public class CSJSONString extends CSJSONType {

    private final String _value;

    public CSJSONString(String value) {
        super(value);
        _value = value;
    }

    public CSJSONString asJSONString() {
        return this;
    }

    public String get() {
        return _value;
    }

}

File: library/renetik-base/src/main/java/renetik/android/framework/json/data/properties/CSJsonString.kt
package renetik.android.framework.json.data.properties

import renetik.android.framework.lang.property.CSProperty
import renetik.android.framework.json.data.CSJsonObject

class CSJsonString(val data: CSJsonObject, private val key: String)
    : CSProperty<String>, CharSequence {
    var string: String?
        get() = data.getString(key)
        set(value) {
            data.set(key, value)
        }

    override var value
        get() = string ?: ""
        set(value) {
            string = value
        }

    override fun toString() = value
    override val length get() = value.length
    override fun get(index: Int) = value[index]
    override fun subSequence(startIndex: Int, endIndex: Int) =
        value.subSequence(startIndex, endIndex)

    fun value(value: String) {
        string = value
    }
}